networkx/algorithms/tests/test_walks.py

```diff
@@ -0,0 +1,64 @@
+"""Unit tests for the :mod:`networkx.algorithms.walks` module."""
+
+import pytest
+
+import networkx as nx
+
+pytest.importorskip("numpy")
+pytest.importorskip("scipy")
+
+
+class TestNumberOfWalks:
+    @classmethod
+    def setup_class(cls):
+        cls.G = nx.DiGraph(nx.utils.pairwise(range(3), cyclic=True))
+
+    def test_neither_source_nor_target(self):
+        num_walks = nx.number_of_walks(self.G, 3)
+        expected = {0: {0: 1, 1: 0, 2: 0}, 1: {0: 0, 1: 1, 2: 0}, 2: {0: 0, 1: 0, 2: 1}}
+        assert num_walks == expected
+
+    def test_source_only(self):
+        num_walks = nx.number_of_walks(self.G, 2)
+        expected = {0: 0, 1: 1, 2: 0}
+        assert num_walks[2] == expected
+
+    def test_source_and_target(self):
+        num_walks = nx.number_of_walks(self.G, 2)
+        assert num_walks[0][2] == 1
```

Comment:
Now that we're down to only one function `nx.number_of_walks`, these test cases are redundant as the `test_neither_source_nor_target` covers them. The only difference is how the resultant dictionary is accessed, which doesn't really test the behavior of `nx.number_of_walks`.

This goes for some of the tests below as well, which are covered by `test_basic`.